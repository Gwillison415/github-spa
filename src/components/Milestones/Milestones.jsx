import React, { Component } from 'react';
import { Card } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { retrieveMilestones } from '../../actions/milestonesActions';
import Milestone from '../Milestone/Milestone';
import './Milestones.css';

export class Milestones extends Component {
  componentDidMount() {
    this.props.retrieveMilestones();
  }

  milestoneComponents(milestonesIds) {
    return milestonesIds.map(id => (
      <Milestone key={id} milestoneId={id} />
    ));
  }

  render() {
    if (this.props.loadingMilestones) {
      return (
        <div>Loading Milestones</div>
      );
    }
    return (
      <Card>
        <Card.Content>
          <Card.Header className="ui center aligned">
            Milestones
          </Card.Header>u
        </Card.Content>
        {this.milestoneComponents(this.props.milestonesIds)}
      </Card>
    );
  }
}

Milestones.propTypes = {
  retrieveMilestones: PropTypes.func.isRequired,
  loadingMilestones: PropTypes.bool,
  milestonesIds: PropTypes.arrayOf(PropTypes.number).isRequired,
};

Milestones.defaultProps = {
  loadingMilestones: false,
};

export const mapStateToProps = state => ({
  milestonesIds: state.milestones.ids,
  milestonesById: state.milestones.milestonesById,
  loadingMilestones: state.loadingMilestones,
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators({
    retrieveMilestones,
  }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Milestones);
