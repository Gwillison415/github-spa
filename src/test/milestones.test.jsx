import React from 'react';
import toJson, { shallowToJson } from 'enzyme-to-json';
import { shallow, render, mount } from 'enzyme';
import { Milestones, mapStateToProps, mapDispatchToProps } from '../components/Milestones/Milestones';
import retrieveMilestones from '../actions/milestonesActions';

const state = {
  milestones: {
    ids: [2601714],
  },
  milestonesById: {
    2601714: {
      created: '2017-06-23T22:45:34Z',
      labels: [],
      repoIssueNumber: 36,
      title: 'disregard this PR',

    },
  },
  loadingMilestones: false,
  currentPage: { userName: 'michaelmurray6298' },
};

describe('Milestones component', () => {
  it('should render a component with props as specified ', () => {
    const component = shallow(
      <Milestones
        loadingMilestones={false}
        retrieveMilestones={retrieveMilestones}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('map\'s given State To Props ', () => {
    const expected = {
      loadingMilestones: false,
      userName: 'michaelmurray6298',
      orgName: undefined,
      milestonesById: undefined,
      milestonesByRepo: undefined,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('maps component dispatches to props', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('retrieveMilestones');
  });
  it('Should have a Cart Header in the center', () => {
    const component = shallow(
      <Milestones
        loadingMilestones={false}
        retrieveMilestones={retrieveMilestones}
        userName={state.currentPage.userName}
      />,
    );
    expect(component.find('.aligned').exists()).toBe(true);
  });
});
