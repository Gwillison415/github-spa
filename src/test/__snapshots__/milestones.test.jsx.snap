// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Milestones component should render a component with props as specified  1`] = `
<Card>
  <CardContent>
    <CardHeader
      className="ui center aligned"
    >
      Milestones
    </CardHeader>
  </CardContent>
<<<<<<< HEAD
  <CardContent>
    <Feed>
      <List
        divided={true}
        relaxed={true}
      >
        <Connect(Component)
          milestoneId={2601714}
        />
      </List>
    </Feed>
  </CardContent>
=======
  <Connect(Component)
    milestoneId={2601714}
  />
>>>>>>> 02a66ea... Adding with some linting errors addressed
</Card>
`;
